pkg/clients/docker.go

```diff
@@ -1,17 +1,37 @@
 package clients
 
 import (
+	"errors"
 	docker "github.com/fsouza/go-dockerclient"
 	logging "github.com/op/go-logging"
 )
 
-func NewDocker(log *logging.Logger) error {
-	dockerClient, err := docker.NewClient(DockerSocket)
+func Docker(log *logging.Logger) (*docker.Client, error) {
+	errMsg := "Something went wrong initializing Docker client!"
+	once.Docker.Do(func() {
+		client, err := newDocker(log)
+		if err != nil {
+			log.Error(errMsg)
+			log.Error(err.Error())
+			instances.Docker = clientResult{nil, err}
+		}
+		instances.Docker = clientResult{client, nil}
```

Comment:
I tried to reorg these into something generic. It eventually just became the once method with a bunch of type conversions. Almost no gain there, it's error handling boilerplate for the most part that would be required anyway, so I abandoned it. :/